### PR TITLE
feat(authdialog): feature flag rollout

### PR DESCRIPTION
**Pull request by dzucconi**
This looks ready to go — passing against the existing Integrity specs, but will wait until Monday morning to merge and begin rollout. Shoutout Integrity for catching a missing bit of tracking.

### CODE REVIEW

File: src/Components/NavBar/NavBarLoggedOutActions.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+              },
+            },
+            legacy: {
+              mode: ModalType.login,
+              intent: Intent.login,
+              contextModule: ContextModule.header,
+              copy: "Log in to collect art by the world’s leading artists",
+            },
+          })
+        }}
```
Comment (dzucconi) [inline src/Components/NavBar/NavBarLoggedOutActions.tsx:33]: cc @damassi — I'm a little torn about this. I want to break the existing API for two reasons:

* The flat arguments are confusing to me. Between what's used for analytics and what is a configuration option for the actual dialog. Like right now `intent` is used specifically for analytics vs `afterSignUpAction/afterAuthAction`, which actually runs an [`AuthIntent`](https://github.com/artsy/force/blob/main/src/Utils/Hooks/useAuthIntent/index.tsx)
* The way title works is slightly different, accepting an optional function which can be used to configure the string based on view mode. (The existing production implementation actually breaks, where it shows the initial copy string, you change views and it switches it to a generic, then if you swap back to the other view it uses the default for that view.)

Comment (damassi) [reply src/Components/NavBar/NavBarLoggedOutActions.tsx:33]: Updated API looks much better to me assuming there isn't fallout elsewhere 👍

File: src/Components/AuthDialog/useAuthDialog.tsx [hunk 2]
```diff
@@ -8,3 +12,33 @@ export const useAuthDialog = () => {
...
 }
+
+const FEATURE_FLAG_NAME = "auth-dialog"
+
+/**
+ * Returns a function to show the auth dialog; accepting props for both
+ * the new and legacy versions, and choosing which to use based on the
+ * feature flag.
+ */
+export const useAuthDialogAdapter = () => {
```
[the remark this answers is not in the document]
Comment (dzucconi) [reply src/Components/AuthDialog/useAuthDialog.tsx:23]: Yeah, when it's completely rolled out I'd switch this to just the `useAuthDialog` function and clean up the props; but I might as well avoid the import/bulk re-naming.

File: src/Components/AuthDialog/Hooks/useAfterAuthenticationRedirectUrl.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const { loginPagePath, signupPagePath } = getENV("AP")
+
+  // If we're on the login or sign up path; we should redirect to the default (index).
+  // Otherwise stay on the same page.
+  return [loginPagePath, signupPagePath].includes(window.location.pathname)
+    ? DEFAULT_AFTER_AUTH_REDIRECT_PATH
+    : window.location.href
+}
+
+const DEFAULT_AFTER_AUTH_REDIRECT_PATH = "/"
```
[the remark this answers is not in the document]
Comment (dzucconi) [reply src/Components/AuthDialog/Hooks/useAfterAuthenticationRedirectUrl.ts:51]: It's well tested under the hook that wraps it, but I might as well.

[the remark this answers is not in the document]
Comment (damassi) [reply src/Components/AuthDialog/Hooks/useAfterAuthenticationRedirectUrl.ts:51]: Ah, then either way is fine with me 👍 The amount of logic did jump out tho.

**Review by damassi — DISMISSED** [review]
One suggestion to add some tests to that one hook with all of the logic but otherwise looks great

File: src/Components/AuthDialog/useAuthDialog.tsx [hunk 4]
```diff
@@ -1,10 +1,45 @@
...
  * Exposes specific dialog functions and hides internal context.
  */
-export const useAuthDialog = () => {
+export const __useAuthDialog__ = () => {
   const { showAuthDialog, hideAuthDialog } = useAuthDialogContext()
 
   return { showAuthDialog, hideAuthDialog }
 }
+
+const FEATURE_FLAG_NAME = "auth-dialog"
```
Comment (dzucconi) [inline src/Components/AuthDialog/useAuthDialog.tsx:16]: https://unleash.artsy.net/projects/default/features/auth-dialog

**Pull request merged**